### PR TITLE
Allow the use of paths like ~/.project.json

### PR DESCRIPTION
**Pull request by jayme-github**
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /home/janis/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

This makes it easy to manage credentials files outside of the repo (and/or use user specific credentials). For example:

``` yaml
      vars:
         service_account_email: unique-id@developer.gserviceaccount.com
         credentials_file: ~/.project.json
         project_id: project-id
         machine_type: n1-standard-1
         image: debian-7
```

### CODE REVIEW

**Comment by jayme-github** [comment]
Updated PR to reflect changes from #15924 .

**Comment by tback** [comment]
What is blocking this? It seems simple to me and it's unlikely that it'll break anything.

File: lib/ansible/module_utils/gcp.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def _validate_credentials_file(module, credentials_file, require_valid_json=True
     """
     try:
         # Try to read credentials as JSON
-        with open(credentials_file) as credentials:
+        with open(os.path.expanduser(credentials_file)) as credentials:
```
Comment (bcoca) [inline lib/ansible/module_utils/gcp.py]: this is the wrong fix, set the type=path in argspec and it will also resolve issus like $HOME/file.json

Comment (tback) [reply lib/ansible/module_utils/gcp.py]: Tried that, didn't work! How is this supposed to work? I've never heard of python doing magic on argspecs.

Comment (bcoca) [reply lib/ansible/module_utils/gcp.py]: no magic, type 'path' is run via expanduser AND expandvars, it is part of the AnsibleModule class. Can you define 'didn't work' as that would affect many modules?

Comment (tback) [reply lib/ansible/module_utils/gcp.py]: You don't want to know what I thought. Fixed it now.

**Comment by tback** [comment]
Everything should be fine now. Please let me know if you need anything else for this to pass.

**Comment by supertom** [comment]
Thanks for the PR!  (Sorry I missed this, it looks like it didn't have the 'gce' label.)  LGTM!

**Comment by tback** [comment]
Do you mind to merge it then?

**Pull request merged**